src/Orleans/Messaging/SqlMembershipTable.cs

```diff
@@ -21,489 +21,229 @@ MIT License
         private TimeSpan maxStaleness;
+        private TraceLogger logger;
+        private IRelationalStorage database;
+
 
         public async Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
         {
+            logger = traceLogger;
             deploymentId = config.DeploymentId;
-            connectionString = config.DataConnectionString;
 
+            logger.Verbose3("SqlMembershipTable.InitializeMembershipTable called.");
+                                                
+            //TODO: Orleans does not yet provide the type of database used (to, e.g., to load dlls), so SQL Server is assumed.
+            database = RelationalStorageUtilities.CreateGenericStorageInstance(WellKnownRelationalInvariants.SqlServer, config.DataConnectionString);
+
+            //This initializes all of Orleans operational queries from the database using a well known view
+            //and assumes the database with appropriate defintions exists already.
+            await InitializeOrleansQueriesAsync();
+            
             // even if I am not the one who created the table, 
             // try to insert an initial table version if it is not already there,
             // so we always have a first table version row, before this silo starts working.
-            if (tryInitTableVersion)
+            if(tryInitTableVersion)
             {
-                await InitTable();
+                var wasCreated = await InitTableAsync();
+                if(wasCreated)
+                {
+                    logger.Info("Created new table version row.");
+                }
             }
         }
 
+
         public Task InitializeGatewayListProvider(ClientConfiguration config, TraceLogger traceLogger)
         {
-            deploymentId = config.DeploymentId;
-            connectionString = config.DataConnectionString;
+            logger.Verbose3("SqlMembershipTable.InitializeGatewayListProvider called.");
+
+            deploymentId = config.DeploymentId;            
             maxStaleness = config.GatewayListRefreshPeriod;
+
+            //TODO: Orleans does not yet provide the type of database used (to, e.g., to load dlls), so SQL Server is assumed.
+            database = RelationalStorageUtilities.CreateGenericStorageInstance(WellKnownRelationalInvariants.SqlServer, config.DataConnectionString);
+
             return TaskDone.Done;
         }
 
+
         public TimeSpan MaxStaleness
         {
             get { return maxStaleness; }
         }
 
+
         public bool IsUpdatable
         {
             get { return true; }
         }
 
+
         public IList<Uri> GetGateways()
         {
-            var result = new List<Uri>();
-
-            // NOTE: This method is only called from client, so blocking operations should be ok.
-
-            using (var conn = new SqlConnection(connectionString))
+            logger.Verbose3("SqlMembershipTable.GetGateways called.");
```

Comment:
I added an explicit print of calling membership methods. It doesn't feel like there's harm done (performancewise or otherwise), especially on level three of verbosity.